File: JavaAndroidTest/app/src/main/java/taipt4/kotlin/javaandroidtest/MainActivity.java
package taipt4.kotlin.javaandroidtest;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager2.widget.ViewPager2;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.widget.Toast;

import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private String[] animalsArray = {"Cat", "Dog", "Bird"};
    private Toolbar toolbar;
    private TabLayout tabLayout;
    private ViewPager2 viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        toolbar = findViewById(R.id.toolbar);
        viewPager = findViewById(R.id.viewPager);
        tabLayout = findViewById(R.id.tabLayout);

        AnimalsAdapter adapter = new AnimalsAdapter(getSupportFragmentManager(), getLifecycle());
        viewPager.setAdapter(adapter);
        new TabLayoutMediator(tabLayout, viewPager, (tab, position) -> tab.setText(animalsArray[position])).attach();
        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                if (tab.getPosition() == 1) {
                    showDialog();
                }
                Toast.makeText(getApplicationContext(), "Tab" + tab.getText() + " selected", Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {
                Toast.makeText(getApplicationContext(), "Tab" + tab.getText() + " unselected", Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {
                Toast.makeText(getApplicationContext(), "Tab" + tab.getText() + " reselected", Toast.LENGTH_SHORT).show();
            }
        });
    }



    public void showDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setMessage("Test")
                .setPositiveButton("OK", (dialog, id) -> dialog.dismiss())
                .setNegativeButton("CANCEL", (dialog, id) -> dialog.dismiss());
        builder.create().show();
    }
}

File: Example2/app/src/main/java/taipt4/kotlin/example2/MainActivity.kt
package taipt4.kotlin.example2

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var btnStart: Button
    private lateinit var btnStop: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnStart = findViewById(R.id.startButton)
        btnStop = findViewById(R.id.stopButton)

        btnStart.setOnClickListener(this)
        btnStop.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when(v) {
            btnStart -> {
                startService(Intent(this, NewService::class.java))
                Toast.makeText(this, "Start Service", Toast.LENGTH_LONG).show()
            }
            btnStop -> {
                stopService(Intent(this, NewService::class.java))
                Toast.makeText(this, "Stop Service", Toast.LENGTH_LONG).show()
            }
        }
    }
}